diff -ruN a/buildroot/package/cmake/0001-rename-cmake-rootfile.patch b/buildroot/package/cmake/0001-rename-cmake-rootfile.patch
--- a/buildroot/package/cmake/0001-rename-cmake-rootfile.patch	2024-05-08 21:21:13.000000000 +0800
+++ b/buildroot/package/cmake/0001-rename-cmake-rootfile.patch	2026-07-16 03:19:23.000000000 +0800
@@ -12,7 +12,7 @@
 
 [Vincent: tweak patch for 3.6.3]
 [James: tweak patch for 3.15.4]
-[Bernd: tweak patch for 3.27.0]
+[Bernd: tweak patch for 3.27.0 & 3.31.2]
 
 Signed-off-by: Davide Viti <zinosat@tiscali.it>
 Signed-off-by: Vicente Olivert Riera <Vincent.Riera@imgtec.com>
@@ -26,7 +26,7 @@
 index 3772f09985..c317e4af38 100644
 --- a/Source/cmake.cxx
 +++ b/Source/cmake.cxx
-@@ -1891,7 +1891,12 @@ int cmake::AddCMakePaths()
+@@ -1896,7 +1896,12 @@ int cmake::AddCMakePaths()
                        cmStateEnums::INTERNAL);
  #endif
    if (!cmSystemTools::FileExists(
diff -ruN a/buildroot/package/cmake/cmake.hash b/buildroot/package/cmake/cmake.hash
--- a/buildroot/package/cmake/cmake.hash	2024-05-08 21:21:13.000000000 +0800
+++ b/buildroot/package/cmake/cmake.hash	2026-07-16 03:19:23.000000000 +0800
@@ -1,5 +1,5 @@
-# From https://cmake.org/files/v3.28/cmake-3.28.1-SHA-256.txt
-sha256  15e94f83e647f7d620a140a7a5da76349fc47a1bfed66d0f5cdee8e7344079ad  cmake-3.28.1.tar.gz
+# From https://cmake.org/files/v3.31/cmake-3.31.12-SHA-256.txt
+sha256  5f3fd5a54dfa65602bdbed64f981a72673cc19f2d304cc2955cf0dfa0cfd8272  cmake-3.31.12.tar.gz
 
 # Locally calculated
-sha256  4a01ccf2dc580ba570d02bc015bbe0ec92f1f318717aae9540ba841ba7946756  Copyright.txt
+sha256  0676e30827684487aa0ea8f72f01e9460a6320fa34a2888df0ad1dd3e8a06eb8  Copyright.txt
diff -ruN a/buildroot/package/cmake/cmake.mk b/buildroot/package/cmake/cmake.mk
--- a/buildroot/package/cmake/cmake.mk	2024-05-08 21:21:13.000000000 +0800
+++ b/buildroot/package/cmake/cmake.mk	2026-07-16 03:19:23.000000000 +0800
@@ -5,14 +5,15 @@
 ################################################################################
 
 # When updating the version, please also update BR2_HOST_CMAKE_AT_LEAST_X_Y
-CMAKE_VERSION_MAJOR = 3.28
-CMAKE_VERSION = $(CMAKE_VERSION_MAJOR).1
+CMAKE_VERSION_MAJOR = 3.31
+CMAKE_VERSION = $(CMAKE_VERSION_MAJOR).12
 CMAKE_SITE = https://cmake.org/files/v$(CMAKE_VERSION_MAJOR)
 CMAKE_LICENSE = BSD-3-Clause
 CMAKE_LICENSE_FILES = Copyright.txt
 CMAKE_CPE_ID_VALID = YES
-# Tool download MITM attack warning if using npm package to install cmake
-CMAKE_IGNORE_CVES = CVE-2016-10642
+
+# The package is a dependency to ccache so ccache cannot be a dependency
+HOST_CMAKE_ADD_CCACHE_DEPENDENCY = NO
 
 # CMake is a particular package:
 # * CMake can be built using the generic infrastructure or the cmake one.
diff -ruN a/buildroot/package/cmake/Config.in.host b/buildroot/package/cmake/Config.in.host
--- a/buildroot/package/cmake/Config.in.host	2024-05-08 21:21:13.000000000 +0800
+++ b/buildroot/package/cmake/Config.in.host	2026-07-16 03:19:23.000000000 +0800
@@ -47,10 +47,30 @@
 	bool
 	select BR2_HOST_CMAKE_AT_LEAST_3_26
 
+config BR2_HOST_CMAKE_AT_LEAST_3_28
+	bool
+	select BR2_HOST_CMAKE_AT_LEAST_3_27
+
+config BR2_HOST_CMAKE_AT_LEAST_3_29
+	bool
+	select BR2_HOST_CMAKE_AT_LEAST_3_28
+
+config BR2_HOST_CMAKE_AT_LEAST_3_30
+	bool
+	select BR2_HOST_CMAKE_AT_LEAST_3_29
+
+config BR2_HOST_CMAKE_AT_LEAST_3_31
+	bool
+	select BR2_HOST_CMAKE_AT_LEAST_3_30
+
 # This order guarantees that the highest version is set, as kconfig
 # stops affecting a value on the first matching default.
 config BR2_HOST_CMAKE_AT_LEAST
 	string
+	default "3.31"	if BR2_HOST_CMAKE_AT_LEAST_3_31
+	default "3.30"	if BR2_HOST_CMAKE_AT_LEAST_3_30
+	default "3.29"	if BR2_HOST_CMAKE_AT_LEAST_3_29
+	default "3.28"	if BR2_HOST_CMAKE_AT_LEAST_3_28
 	default "3.27"	if BR2_HOST_CMAKE_AT_LEAST_3_27
 	default "3.26"	if BR2_HOST_CMAKE_AT_LEAST_3_26
 	default "3.25"	if BR2_HOST_CMAKE_AT_LEAST_3_25
