diff -ruN a/buildroot/package/mali-driver/Config.in b/buildroot/package/mali-driver/Config.in
--- a/buildroot/package/mali-driver/Config.in	2024-05-08 21:21:22.000000000 +0800
+++ b/buildroot/package/mali-driver/Config.in	2026-09-06 16:43:04.250839592 +0800
@@ -5,6 +5,7 @@
 	bool "mali-driver"
 	depends on BR2_aarch64 || BR2_arm
 	depends on BR2_LINUX_KERNEL
+	select BR2_PACKAGE_MESA3D_HEADERS
 	help
 	  This package builds and installs the Linux kernel driver for
 	  the Mali Midgard/Bifrost GPU, compatible with recent Linux
