diff -ru a/buildroot/package/waylandpp/Config.in b/buildroot/package/waylandpp/Config.in
--- a/buildroot/package/waylandpp/Config.in	2024-05-08 21:21:50.000000000 +0800
+++ b/buildroot/package/waylandpp/Config.in	2026-08-28 11:27:26.380793179 +0800
@@ -1,6 +1,6 @@
 config BR2_PACKAGE_WAYLANDPP
 	bool "waylandpp"
-	depends on BR2_PACKAGE_HAS_LIBEGL_WAYLAND
+	depends on BR2_PACKAGE_HAS_LIBEGL
 	depends on BR2_HOST_GCC_AT_LEAST_4_9
 	depends on BR2_TOOLCHAIN_GCC_AT_LEAST_4_9
 	depends on BR2_INSTALL_LIBSTDCPP
@@ -10,9 +10,9 @@
 
 	  https://github.com/NilsBrause/waylandpp
 
-comment "waylandpp needs an OpenGL-EGL/wayland backend, a toolchain w/ gcc >= 4.9, host gcc >= 4.9, C++ support"
+comment "waylandpp needs an OpenGL-EGL backend, a toolchain w/ gcc >= 4.9, host gcc >= 4.9, C++ support"
 	depends on BR2_PACKAGE_WAYLAND
-	depends on !BR2_PACKAGE_HAS_LIBEGL_WAYLAND || \
+	depends on !BR2_PACKAGE_HAS_LIBEGL || \
 		!BR2_HOST_GCC_AT_LEAST_4_9 || \
 		!BR2_TOOLCHAIN_GCC_AT_LEAST_4_9 || \
 		!BR2_INSTALL_LIBSTDCPP
